app/code/Magento/Wishlist/Model/WishlistManagement.php

```diff
@@ -0,0 +1,227 @@
+        \Magento\Wishlist\Model\ItemFactory $itemFactory
+    ) {
+        $this->_wishlistCollectionFactory = $wishlistCollectionFactory;
+        $this->_productRepository = $productRepository;
+        $this->_wishlistFactory = $wishlistFactory;
+        $this->_itemFactory = $itemFactory;
+    }
+
+    /**
+     * Get wishlist collection
+     * @param int $customerId
+     * @return array WishlistData
+     */
+    public function getWishlistForCustomer($customerId)
+    {
+        if (empty($customerId) || !isset($customerId) || $customerId == "") {
+            $message = __('Id required');
+            $status = false;
+            $response[] = [
+                "message" => $message,
+                "status"  => $status
+            ];
+            return $response;
+        } else {
+            $collection =
+                $this->_wishlistCollectionFactory->create()
+                    ->addCustomerIdFilter($customerId);
+
+            $wishlistData = [];
+            foreach ($collection as $item) {
+                $productInfo = $item->getProduct()->toArray();
+                $data = [
+                    "wishlist_item_id" => $item->getWishlistItemId(),
+                    "wishlist_id"      => $item->getWishlistId(),
+                    "product_id"       => $item->getProductId(),
+                    "store_id"         => $item->getStoreId(),
+                    "added_at"         => $item->getAddedAt(),
+                    "description"      => $item->getDescription(),
+                    "qty"              => round($item->getQty()),
+                    "product"          => $productInfo
+                ];
+                $wishlistData[] = $data;
+            }
+            return $wishlistData;
+        }
+    }
+
+    /**
+     * Add wishlist item for the customer
+     * @param int $customerId
+     * @param int $productIdId
+     * @return array|bool
+     *
+     */
+    public function addWishlistForCustomer($customerId, $productId)
+    {
+        if ($productId == null) {
+            $message = __('Invalid product, Please select a valid product');
+            $status = false;
+            $response[] = [
+                "message" => $message,
+                "status"  => $status
+            ];
+            return $response;
+        }
+        try {
+            $product = $this->_productRepository->getById($productId);
+        } catch (Exception $e) {
+            return false;
+        }
+        try {
+            $wishlist = $this->_wishlistFactory->create()
+                ->loadByCustomerId($customerId, true);
+            $wishlist->addNewItem($product);
+            $wishlist->save();
+        } catch (Exception $e) {
+            return false;
+        }
+        $message = __('Item added to wishlist.');
+        $status = true;
+        $response[] = [
+            "message" => $message,
+            "status"  => $status
+        ];
+        return $response;
```

Comment:
there should not be any message returned from the service call. It's responsibility of presentation layer to provide correct message depending on each specific case and UI.
all the state changing operations should return - Void type